software/bootcamp-code/main.py

```diff
@@ -33,61 +109,265 @@ def main_loop():
     # open the serial connection
     omni_motion.open()
     
-    state=State.FIND_BALL
+    # state if to listen to referee commands, when needed, change to True
+    referee_active = True
+    command_list=[]
+    robot_id="twelve"
+
+    # listen for referee commands
+    if referee_active:
+        print("start referee")
+        thread = Thread(target=run_websocket, args=(command_list,))
+        thread.start()
+        state=State.STOP
+    # list for referee commands and robot_id
+    else:
+        # default state to start with
+        state=State.FIND_BALL
+
+    # variables to control driving to furthest basket when ball has not been found for max_find_time
+    finder_timer=time.time()
+    max_find_time=5
+    furthest_basket = BasketColor.BLUE
+    basket_to_drive=None
+    basket_blue_d=0
+    basket_magn_d=0
+    start_opbasket_drive=time.time()
+
+
+    # if want to test the thrower, comment in
+    #state = State.TEST_THROWER
+    testing_thrower_speed=1900
 
     start = time.time()
     fps = 0
     frame = 0
     frame_cnt = 0
-    try:
-        # time counter for testing
-        zero_time = time.time()
-        # Ball desired coordinates are in the middle of the frame.
-        ball_desired_x = cam.rgb_width/2
-        ball_desired_y = cam.rgb_height/2
+    
+    # time counter for testing
+    # zero_time = time.time()
         
-        while True:
-            # has argument aligned_depth that enables depth frame to color frame alignment. Costs performance
-            processedData = processor.process_frame(aligned_depth=False)
+    # Ball desired coordinates are in the middle of the frame width and 400 from the upper edge of the frame
+    # (THE ZERO COORDINATES OF THE FRAME ARE IN THE UPPER LEFT CORNER.)
+    ball_desired_x = cam.rgb_width/2
+    ball_desired_y = 400
```

Comment:
Can be comupted as a ratio to be clearer what the actual value means. cam.rgb_width*0.8 is clearer than a pixel value since it adds the context of the frame height and a percentage to the code